frontend/src/components/Map/index.js

```diff
@@ -7,204 +7,217 @@ import api from "utils";
+    lng: -119.6,
+    lat: 36.7,
   });
 
+  useEffect(() => {
+    getUserLocation();
+
+    const map = new mapboxgl.Map({
+      container: "map",
+      style: "mapbox://styles/mapbox/dark-v10",
+      center: [location.lng, location.lat],
+      zoom: initialZoom,
+    });
+
+    addLayers(map);
+    setMap(map);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    map &&
+      map.flyTo({
+        center: [location.lng, location.lat],
+        zoom: focusZoom,
+        speed: 0.6,
+        curve: 1,
+      });
+  }, [location, map]);
+
   const getUserLocation = async () => {
     const userLocation = await fetchUserLocation();
     if (userLocation) {
-      return {
+      setLocation({
+        ...location,
         lng: userLocation.lng,
         lat: userLocation.lat,
-      };
+      });
     }
   };
 
-  const addLayers = async (pMap) => {
+  const addLayers = (map) => {
     const worldData = fetchCovidData(dataScope.WORLD);
     const usStatesData = fetchCovidData(dataScope.US_STATES);
-    const map = await pMap;
 
     map.on("load", function () {
       addWorldLayer(map, worldData);
       addUSStatesLayer(map, usStatesData);
     });
   };
 
-  useEffect(() => {
-    const initializeMap = async () => {
-      const userLocation = await getUserLocation();
-      const location = userLocation ? userLocation : state.location;
-      return new mapboxgl.Map({
-        container: "map",
-        style: "mapbox://styles/mapbox/dark-v10",
-        center: [location.lng, location.lat],
-        zoom: state.zoom,
-      });
-    };
-
-    const map = initializeMap();
-    addLayers(map);
-  }, []);
+  const fetchUserLocation = async () => {
+    const response = await fetch(`http://ip-api.com/json`);
+    if (response.status >= 200 && response.status < 300) {
+      const jsonResponse = await response.json();
+      return { lat: jsonResponse.lat, lng: jsonResponse.lon };
+    }
+  };
 
-  return (
-    <div className={styles.root}>
-      <div className={classNames(styles.fill)} id="map"></div>
-      {!draggable && <div className={classNames(styles.fill, styles.mask)} />}
-    </div>
-  );
-}
+  const fetchCovidData = async (scope) => {
+    return await api(`data/${scope}`, {
+      method: "GET",
+    });
+  };
 
-async function fetchCovidData(scope) {
-  return await api(`data/${scope}`, {
-    method: "GET",
-  });
-}
+  const getColor = (group) => {
+    return `rgba(${group * 255}, 0, 0, 1)`;
+  };
 
-async function addWorldLayer(map, data) {
-  const covidData = await data;
-  const expression = covidData.map((row) => [row.country, getColor(row.group)]);
-
-  map.addLayer(
-    {
-      id: "maine",
-      type: "fill",
-      layout: {},
-      paint: {
-        "fill-color": {
-          type: "categorical",
-          property: "name",
-          stops: expression,
+  const addWorldLayer = async (map, data) => {
+    const covidData = await data;
```

Comment:
haha for some reason this is hillarious to read and super semantic, `await data` I love it